hazelcast-sql/src/test/java/com/hazelcast/sql/impl/calcite/validate/types/ReplaceUnknownOperandTypeInferenceTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.validate.types;
+
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static junit.framework.TestCase.assertEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class ReplaceUnknownOperandTypeInferenceTest extends InferTypesTestSupport {
```

Comment:
Might be better to use `InferTypesTestSupport` as a utility rather than as a superclass.

Reply:
This is a common pattern that we use throughout IMDG - define "*Support" class and then extend it. I think both approaches are roughly equal - no big win, no big harm wrt each other.

Reply:
It will show if you try to use a method from another support class. Utility classes are more flexible.